### PR TITLE
sudoers: initial integration

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Cross-refencing https://github.com/google/oss-fuzz/issues/5050

I have added two fuzzers for now and am waiting for feedback from the authors.

### CODE REVIEW

**Comment by DavidKorczynski** [comment]
@inferno-chromium @jonathanmetzman @oliverchang This one should be good to go. 

The fail on the CI is due to a valid bug being catched. The root-cause is in this comment https://github.com/sudo-project/sudo/issues/84#issuecomment-768375711 and you can see the confirmation that this is a true positive as the next comment.

**Review by inferno-chromium — APPROVED** [review]
Can you please remind them to fix that null crasher, otherwise fuzzing wont start.

Also, these fuzzer seem to find crash in parsing logic, should be checking if sudo privilege escalation worked in some way ?

**Comment by DavidKorczynski** [comment]
> Can you please remind them to fix that null crasher, otherwise fuzzing wont start.
> 
yes, I can also see if I can submit a patch a bit later today.

> Also, these fuzzer seem to find crash in parsing logic, should be checking if sudo privilege escalation worked in some way ?

Yes, we can extend with more fuzzers and I think the maintainers are happy to start writing fuzzers as well (since this was on their TODO list per the issue in the upstream repo).

File: projects/sudoers/fuzz_sudoers_parse.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+#include "sudoers_version.h"
+#include "sudo_lbuf.h"
+#include "redblack.h"
+#include "cvtsudoers.h"
+#include <gram.h>
+
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+    char filename[256];
+    sprintf(filename, "/tmp/libfuzzer.%d", getpid());
```
Comment (blackgnezdo) [inline projects/sudoers/fuzz_sudoers_parse.c:45]: [mkstemp(3)](https://man7.org/linux/man-pages/man3/mkstemp.3.html) is a much better option. Having /tmp races is a faux pas in a security related code.

**Pull request merged**